packages/core/src/shared/sam/localLambdaRunner.ts

```diff
@@ -307,7 +307,7 @@ export async function runLambdaFunction(
         const payload =
             config.eventPayloadFile === undefined
                 ? undefined
-                : JSON.parse(await readFile(config.eventPayloadFile, { encoding: 'utf-8' }))
+                : JSON.parse(await fs.readFileAsString(config.eventPayloadFile))
```

Comment:
off-topic: we should rename this function :)
```suggestion
                : JSON.parse(await fs.readFileText(config.eventPayloadFile))
```